app/components/emoji/emoji.js

```diff
@@ -49,8 +50,13 @@ export default class Emoji extends React.PureComponent {
             return <Text style={textStyle}>{literal}</Text>;
         }
 
+        let ImageComponent = WebImage;
+        if (Platform.OS === 'android') {
+            ImageComponent = Image;
```

Comment:
We get the dreaded `View component embedded in a Text node` error so we can't use it on Android until that is resolved. I will say that I was only seeing caching issues on iOS. Android cached the emoji's just fine.